### PR TITLE
BUG: setting categorical values into object dtype DataFrame

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

```
df = pd.DataFrame({"A": range(3)}, dtype=object)
cat = pd.Categorical(["alpha", "beta", "gamma"])

df.iloc[range(3), 0] = cat

>>> df
     A
0  NaN
1     
2     

```

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -965,7 +965,13 @@ def setitem(self, indexer, value):
             # GH25495 - If the current dtype is not categorical,
             # we need to create a new categorical block
             values[indexer] = value
-            return self.make_block(Categorical(self.values, dtype=arr_value.dtype))
+            if values.ndim == 2:
+                # TODO(EA2D): special case not needed with 2D EAs
+                if values.shape[-1] != 1:
```
Comment (jreback) [inline pandas/core/internals/blocks.py:970]: `np.atleast_2d` no?

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a whatsnew as well

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -965,7 +965,13 @@ def setitem(self, indexer, value):
             # GH25495 - If the current dtype is not categorical,
             # we need to create a new categorical block
             values[indexer] = value
-            return self.make_block(Categorical(self.values, dtype=arr_value.dtype))
+            if values.ndim == 2:
```
Comment (jreback) [inline pandas/core/internals/blocks.py:968]: you could instead do
```values = np.atleast_2d(values)[:, 0]``` generically i think

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/blocks.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:968]: we dont get a whole lot of mileage out of that bc we still need to (well sort of, it shouldnt be reached) exclude the raising case on L972

**Review by jreback — APPROVED** [review]

**Pull request merged**